dist/counters.d.ts

```diff
@@ -0,0 +1,239 @@
+    timestamp: number[];
+}
+export interface ProcessStatistics {
+    cpu: number;
+    memory: number;
+    ppid: number;
+    pid: number;
+    ctime?: number;
+    elapsed: number;
+    timestamp: number;
+}
+export declare const ProcessStatisticsUnits: any;
+/**
+ * Defines an interface to specify the counters options for counters tests.
+ * @param collectionInterval - the number of milliseconds after which the counters are collected. Default value is provided by environment variable: CountersCollectionInterval and if undefined then by {@link DefaultCountersOptions}.
+ * @param includeMovingAverages - flag if we should compute movingAverages. Default value is provided by environment variable: CountersIncludeMovingAverages and if undefined then by {@link DefaultCountersOptions}.
+ * @param dumpToFile - flag if counters should be dumped to file. Default value is provided by environment variable: CountersDumpToFile and if undefined then by {@link DefaultCountersOptions}.
+ * @param dumpToChart - flag if counters should be dumped to charts. Default value is provided by environment variable: CountersDumpToChart and if undefined then by {@link DefaultCountersOptions}.
+ * @param outputDirectory - provides path to outputDirectory where output files are written. Default value is provided by environment variable: CountersOutputDirectory and if undefined then by {@link DefaultCountersOptions}.
+ */
+export interface CountersOptions {
+    collectionInterval?: number;
+    includeParent?: boolean;
+    includeMovingAverages?: boolean;
+    dumpToFile?: boolean;
+    dumpToChart?: boolean;
+    outputDirectory?: string;
+}
+/**
+ * The default values for CountersOptions.
+ */
+export declare const DefaultCountersOptions: CountersOptions;
+/**
+ * A class with methods that help to implement the counters collection for a test method.
+ */
+export declare class Counters {
+    static readonly MaxCollectionInterval: number;
+    static readonly ProcessInfoUpdationInterval: number;
+    static readonly TotalsProcessInfo: ProcessInfo;
+    /**
+     * the root pid of processes that we are tracking
+     *
+     * @type {number}
+     * @memberof Counters
+     */
+    readonly pid: number;
+    /**
+     * Number of milliseconds interval at which to collect the data.
+     *
+     * @type {number}
+     * @memberof Counters
+     */
+    readonly collectionInterval: number;
+    /**
+     * whether we should include parent of the pid in the process that we are tracking. Including parent implies
+     * we track all of the parent's children and not just the process designated by {@link pid}
+     *
+     * @type {boolean}
+     * @memberof Counters
+     */
+    readonly includeParent: boolean;
+    /**
+     * whether we should dump the counters to file identified by the name of the counter.
+     *
+     * @type {boolean}
+     * @memberof Counters
+     */
+    readonly includeMovingAverages: boolean;
+    readonly dumpToFile: boolean;

```

Comment:
this class has a lot of responsibilities. Maybe instead of having it be aware of where data can be written, you could define separate classes that can implement writing and let the caller create one. EG a "FileDumper" and a "ChartDumper"

That way the settings for charting and file writing can be cleanly separated as well.

Reply:
Good suggestion: However, I do not want incur the cost now. I will keep the suggestion in mind for future work or might refactor this in a future checking.